crates/storage/provider/src/providers/static_file/metrics.rs

```diff
@@ -51,13 +20,23 @@ impl StaticFileProviderMetrics {
         files: usize,
         entries: usize,
     ) {
-        self.segments.get(&segment).expect("segment metrics should exist").size.set(size as f64);
-        self.segments.get(&segment).expect("segment metrics should exist").files.set(files as f64);
-        self.segments
-            .get(&segment)
-            .expect("segment metrics should exist")
-            .entries
-            .set(entries as f64);
+        match segment {
+            StaticFileSegment::Headers => {
+                self.segments.headers_size.set(size as f64);
+                self.segments.headers_files.set(files as f64);
+                self.segments.headers_entries.set(entries as f64);
+            }
+            StaticFileSegment::Transactions => {
+                self.segments.transactions_size.set(size as f64);
+                self.segments.transactions_files.set(files as f64);
+                self.segments.transactions_entries.set(entries as f64);
+            }
+            StaticFileSegment::Receipts => {
+                self.segments.receipts_size.set(size as f64);
+                self.segments.receipts_files.set(files as f64);
+                self.segments.receipts_entries.set(entries as f64);
+            }
+        }
```

Comment:
this part looks good to me, see other comment for ways to make the match arms better

Reply:
thanks